src/ORM/FieldType/DBGeneratedColumn.php

```diff
@@ -0,0 +1,288 @@
+<?php
+
+namespace SilverStripe\ORM\FieldType;
+
+use InvalidArgumentException;
+use SilverStripe\Assets\Storage\DBFile;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Forms\FormField;
+use SilverStripe\Model\ModelData;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\Filters\SearchFilter;
+
+/**
+ * A column which gets generated based on some SQL statement.
+ * This can be stored (generated whenever columns in the generation statement are updated), or virtual (generated on the fly when accessed).
+ */
+class DBGeneratedColumn extends DBField
+{
+    public const string GENERATION_STORED = 'STORED';
+
+    public const string GENERATION_VIRTUAL = 'VIRTUAL';
```

Comment:
```suggestion
    public const string STORED = 'STORED';

    public const string VIRTUAL = 'VIRTUAL';
```

I don't think there's really any need to prefix the consts

Reply:
I will keep the prefixes. This is just subjective - and I prefer to give constants which are intended to be used in the same place a prefix. This makes it easy to see in the IDE that they're related.
Note that this class inherits other constants from superclasses so these aren't the only constants you'll see in your IDE.